src/item_factory.cpp

```diff
@@ -1048,15 +1067,43 @@ void Item_factory::load( islot_comestible &slot, JsonObject &jo )
     } else {
         jo.read( "nutrition", slot.nutr );
     }
+
+    // any specification of vitamins suppresses use of material defaults @see Item_factory::finalize
+    if( jo.has_int( "vitamins" ) ) {
+        // convenience syntax for setting all vitamins concurrently
+        for( auto &v : vitamin::all() ) {
```

Comment:
I mean, use defaults if no "vitamins" (or "relative") is specified, otherwise use array. That way you could define 0 vitamins as `"vitamins" : [ ]`, without the `int` shorthand having to exist.